### ISSUE TITLE
Avoid units in `poliastro.core`

### ISSUE THREAD

**Issue by jorgepiloto** [poliastro/poliastro#1256]
While reviewing @Yash-10  work in #1254, I noticed that the `astropy.units` are being used within the `core` sub-package by the [atmospheric_drag_model function](https://github.com/poliastro/poliastro/blob/3acb5c8b843786fa874f0ea5b620382cd26ac77f/src/poliastro/core/perturbations.py#L131), see:

```python
rho = model.density((H - R) * u.km).to(u.kg / u.km ** 3).value
```

This line of code breaks one of the key points about the `core` sub-package, which is letting user to manage its own units without requiring astropy ones.

The reasons why I claim a refactor on the atmosphere are listed below these lines:

* We can solve for the issue presented before by moving the logic of the atmospheric models to the core.
* Taking advantage of this shipping, we could `jit` the atmospheric functions by refactoring the class approach into a pure function one.

This is tedious task but otherwise, units are being used within core. I would like other project devs and users to discuss a little bit more about this before working on it.

**Comment by jorgepiloto** [comment]
Ping @astrojuanlu

**Comment by Yash-10** [comment]
Hi @jorgepiloto, it was interesting to note that `astropy.units` are being used in `core`. Moving atmospheric computations to core by `jit`ting them might really improve the performance significantly since there are a lot of for loops and numerical values throughout the `earth/atmosphere` package. For example, the module: [earth/atmosphere/util.py](https://github.com/poliastro/poliastro/blob/main/src/poliastro/earth/atmosphere/util.py) could be immediately allowed to use floats. As you say, this transformation is a bit cumbersome, I had tried my hand on it once but couldn't get too far.

Also, would there be an option to `jit` the individual classes altogether?

**Comment by astrojuanlu** [comment]
Woops, you're right. I agree with @jorgepiloto , we should make this code unitless. I propose that we add a rule that only `numpy` and standard library can be imported in `poliastro.core`:

https://pypi.org/project/import-linter/

@Yash-10 I think what you propose is a different thing than making `atmospheric_drag_model` unitless, am I right? If so, could you please open a new issue? And if not, would you care to explain how it relates? :) In any case, I'm +1 on moving the `atmosphere/util` functions to `core` as well.

About `@jitclass`, if it works, go ahead! However I think `@jitclass` has some limitations, that's why I haven't tried to use it more widely yet. But it's a good moment to start.

**Comment by Yash-10** [comment]
Sorry for the late reply here. I realized I was talking more about `jit`ting the core computation's instead of making existing `core` unitless.

**Comment by astrojuanlu** [comment]
While working on this, I got blocked by probably the same problems @jorgepiloto mentioned here https://github.com/poliastro/poliastro/issues/1302#issuecomment-938404857 I'm going to solve them the quick 'n' dirty way. PR incoming.
